crates/bitwarden-auth/src/registration.rs

```diff
@@ -5,12 +5,45 @@
 //! authentication method such as SSO or master password, and a decryption method such as
 //! key-connector, TDE, or master password.
 
-use bitwarden_core::Client;
+use bitwarden_api_api::models::{
+    DeviceKeysRequestModel, KeysRequestModel, OrganizationUserResetPasswordEnrollmentRequestModel,
+};
+use bitwarden_core::{
+    Client, OrganizationId, UserId,
+    key_management::account_cryptographic_state::WrappedAccountCryptographicState,
+};
+use bitwarden_encoding::B64;
+use bitwarden_error::bitwarden_error;
+use thiserror::Error;
+use tracing::info;
 #[cfg(feature = "wasm")]
 use wasm_bindgen::prelude::*;
 
+/// Request parameters for TDE (Trusted Device Encryption) registration.
+#[cfg_attr(
+    feature = "wasm",
+    derive(tsify::Tsify),
+    tsify(into_wasm_abi, from_wasm_abi)
+)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub struct TdeRegistrationRequest {
+    /// Organization ID to enroll in
+    pub org_id: OrganizationId,
+    /// Organization's public key for encrypting the reset password key. This should be verified by
+    /// the client and not verifying may compromise the security of the user's account.
+    pub org_public_key: B64,
+    /// User ID for the account being initialized
+    pub user_id: UserId,
+    /// Device ID for TDE enrollment
+    pub device_id: String,
```

Comment:
```suggestion
    /// Device identifier for TDE enrollment
    pub device_identifier: String,
```

We have several entities that have both an `.Id` and `.Identifier`, device is one of them. I would suggest making it clear which one we are using.

Reply:
Applied, nice catch!